src/cursor-context.ts

```diff
@@ -86,4 +96,24 @@ function getTodoContext(match: RegExpExecArray, cursorPos: Position): IContextDa
         line,
         range
     }
+}
+
+function getCheckboxContext(match: RegExpExecArray, cursorPos: Position): IContextData {
+    const line = cursorPos.line;
+
+    const prefix = match[1];
+    const checkbox = match[2];
+
+    const start = match.index + match[1].length;
+    const startPos = new Position(line, start);
+    const end = start + checkbox.length;
+    const endPos = new Position(line, end);
+    const range = new Range(startPos, endPos);
+
+    return {
+        data: checkbox,
+        dataLabel: CHECKBOX,
+        line,
+        range
+    }
```

Comment:
There is a lot of code duplication between `getTodoContext` and `getCheckboxContext`. I think it would make sense to have two functions

```
getMatchContext(match, cursorPos, datalabel) // for regexps where data is in the first group
getPrefixedMatchContext(match, cursorPos, datalabel) // for regexps where data is in the second group
```

and use those from `getTimestampContext`, `getTodoContext` and `getCheckboxContext`.